Cabal/Distribution/Simple/PackageIndex.hs

```diff
@@ -435,12 +464,42 @@ lookupPackageName index name =
 -- We get back any number of versions of the specified package name, all
 -- satisfying the version range constraint.
 --
+-- This does NOT work for internal dependencies, DO NOT use this
+-- function on those; use 'lookupInternalDependency' instead.
+--
 -- INVARIANT: List of eligible 'IPI.InstalledPackageInfo' is non-empty.
 --
 lookupDependency :: InstalledPackageIndex -> Dependency
                  -> [(Version, [IPI.InstalledPackageInfo])]
 lookupDependency index (Dependency name versionRange) =
```

Comment:
Also perhaps that can mean using only the internal on in configure to save a little code as I did before. I bring it up because with LibDependency it will work that way anyways.